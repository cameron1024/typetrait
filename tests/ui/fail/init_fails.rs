fn main() {
    let enabled = Enabled;
}

fn foo() -> Enabled {
    unimplemented!()
}

typetrait::union! {
    pub Status = Enabled | Disabled
}


