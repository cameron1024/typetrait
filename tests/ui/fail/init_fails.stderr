error[E0423]: expected value, found enum `Enabled`
  --> tests/ui/fail/init_fails.rs:2:19
   |
2  |     let enabled = Enabled;
   |                   ^^^^^^^
   |
note: the enum is defined here
  --> tests/ui/fail/init_fails.rs:9:1
   |
9  | / typetrait::union! {
10 | |     pub Status = Enabled | Disabled
11 | | }
   | |_^
   = note: this error originates in the macro `typetrait::union` (in Nightly builds, run with -Z macro-backtrace for more info)
